File: Human_written_code/Code_with_comments/Python/Q20.py
def myPow(x: float, n: int) -> float:
    ans = 1.0
    nn = n
    if nn < 0:
        nn = -nn  # Handle negative exponent by converting it to positive

    while nn:
        if nn % 2:
            ans *= x  # If odd, multiply result by x
            nn -= 1
        else:
            x *= x  # Square the base
            nn //= 2  # Divide the exponent by 2

    if n < 0:
        ans = 1.0 / ans  # Invert the result for negative exponent

    return ans

if __name__ == "__main__":
    print(myPow(2, 10))  # Example usage

File: Human_written_code/Code_with_comments/CPP/Q20.cpp
#include<bits/stdc++.h>

using namespace std;

// Function to calculate x raised to the power n
double myPow(double x, int n) {
  double ans = 1.0;  // Initialize result
  long long nn = n;  // Handle negative power case

  if (nn < 0) nn = -1 * nn;  // Convert to positive for easier handling

  while (nn) {
    if (nn % 2) {
      ans = ans * x;  // Multiply result by x if n is odd
      nn = nn - 1;    // Decrease n
    } else {
      x = x * x;      // Square the base if n is even
      nn = nn / 2;    // Halve the exponent
    }
  }
  
  if (n < 0) ans = 1.0 / ans;  // Invert result if original n was negative

  return ans;  // Return the final result
}

int main() {
  cout << myPow(2, 10) << endl;  // Example usage
}
